### PR TITLE
Tornado 5.0 support

### PR DESCRIPTION
**Pull request by noamkush**

### CODE REVIEW

File: momoko/connection.py [hunk 1]
```diff
@@ -30,6 +30,12 @@
 from tornado.ioloop import IOLoop
 from tornado.concurrent import chain_future, Future
 
+try:
+    from tornado.concurrent import future_set_exc_info
+except ImportError:
+    def future_set_exc_info(future, exc_info):
+        future.set_exc_info(exc_info)
```
Comment (MrShiny608) [inline momoko/connection.py]: This line assumes the future is a `Tornado.concurrent.future`, but [this line](https://github.com/tornadoweb/tornado/blob/master/tornado/concurrent.py#L370) suggests that the Future can be an `asyncio.Future` (as is my case), so `.set_exc_info()` does not exist, you probably want to check what type of future this is an use [.set_exception()](https://docs.python.org/3/library/asyncio-task.html#asyncio.Future.set_exception) if it is an asyncio one.

**Comment by haizaar** [comment]
Hi Noam,

Please make sure the unittests are passing first.

@MrShiny608 raised quite a valid concern on asynio compatibility (I'm not sure why he seems to have deleted his comment). I think it has to be addressed, particularly since Tornado 5.0 focuses on asyncio compatibility.

P.S. I've also updated Travis matrix as per your suggestion. Thank you for that.

**Comment by noamkush** [comment]
I tried looking into the failing test and it looks like the test relies on the old add_done_callback implementation, but I'm not familiar enough with the code to fix the test. I'm not sure it's easily fixable at all. I've spent a few hours on this with no success.
Regarding asyncio compatibility, I don't think the comment was correct (and I'm guessing MrShiny608 realized that after commenting), if you take a look at tornado's [`future_set_exc_info`](https://github.com/tornadoweb/tornado/blob/branch5.0/tornado/concurrent.py#L584) implementation.

**Comment by noamkush** [comment]
That just means you haven't reached any of the set_exc_info uses

**Comment by noamkush** [comment]
It would have been nicer to merge this with credit but thanks anyway.

**Comment by haizaar** [comment]
You code wan't merge-able - didn't pass tests, didn't cater for other things like setup.py, etc. But it definitely provided inspiration, so thank you for that.

**Pull request closed without merging**